tycho-artifactcomparator/src/main/java/org/eclipse/tycho/zipcomparator/internal/TextComparator.java

```diff
@@ -36,62 +36,59 @@ public class TextComparator implements ContentsComparator {
 
     static final String HINT = "txt";
 
+    private static final char CR = '\r';
+    private static final char LF = '\n';
+
+    // Possible new lines:
+    // \n -- unix style
+    // \r\n -- windows style
+    // \r -- whatever style (ignored?!)
```

Comment:
this is macos https://stackoverflow.com/questions/38688516/line-break-represented-as-r-on-mac

Reply:
According to [Wikipedia's Mac OS 9 article](https://en.wikipedia.org/wiki/Mac_OS_9), the latest release of Mac OS 9 is 9.2.2 / December 5, 2001, which was 21 years ago. Therefore I think this is not really relevant anymore.

Nevertheless my intention was that this works as if one would use a `BufferedReader` to read the lines one by one using BufferedReader.readLine() and compare the line content, just more efficient (i.e. without creating Strings all the time).
And the javadoc of BufferedReader.readLine() says:
```
Reads a line of text. A line is considered to be terminated by any oneof a line feed ('\n'), a carriage return ('\r'),
a carriage return followed immediately by a line feed, or by reaching the end-of-file(EOF).
```
So if equivalence is the goal, `\r` needs to be considered.